Matveibos/hometask-3/rainfall_stat.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+# this class is used for calculate mean and variance values for town rainfall
+class RainfallStat
+  SEPARATOR = ':'
+  CHOOSING_NUMBERS = /\d*\.\d/.freeze
+
+  def initialize(town, data)
+    @town = town
+    @data = data
+    @processed_data = {}
+    create_hash
+  end
+
+  def mean
+    calc_mean
+  end
+
+  def variance
+    calc_variance
+  end
+
+  private
+
+  def create_hash
+    @data.each_line do |str|
+      key = str.split(SEPARATOR).first
+      array = str.scan(CHOOSING_NUMBERS).map(&:to_f)
```

Comment:
I remember leaving comment to rename `key` and `array` to something more meaningful